conans/client/generators/json_generator.py

```diff
@@ -71,3 +51,31 @@ def get_options(self):
             for key, value in self.conanfile.options[req].items():
                 options[req][key] = value
         return options
+
+    def serialize_cpp_info(self, depname, cpp_info):
+        keys = [
+            "version",
+            "description",
+            "rootpath",
+            "sysroot",
+            "include_paths", "lib_paths", "bin_paths", "build_paths", "res_paths",
+            "libs",
+            "system_libs",
+            "defines", "cflags", "cxxflags", "sharedlinkflags", "exelinkflags",
+            "frameworks", "framework_paths", "names", "filenames",
+            "build_modules", "build_modules_paths"
+        ]
+        res = {}
+        for key in keys:
+            res[key] = getattr(cpp_info, key)
+        res["cppflags"] = cpp_info.cxxflags  # Backwards compatibility
+        res["name"] = depname
+
+        # FIXME: trick for NewCppInfo objects when declared layout
+        if cpp_info.version is None:
```

Comment:
Maybe put the ``cpp_info.version`` inside the try-except, just in case ``version`` is not defined for NewCppInfo?